kubeflow-katib.yaml

```diff
@@ -0,0 +1,80 @@
+package:
+  epoch: 0
+  name: kubeflow-katib
+  version: 0.15.0
+  description: Kubeflow Katib services
+  copyright:
+    - license: Apache-2.0
+  dependencies:
+    runtime:
+      - wget
+      - ca-certificates
+
+environment:
+  contents:
+    packages:
+      - busybox
+      - ca-certificates-bundle
+      - go
+      - build-base
+      - python-3.10-dev
+      - py3.10-pip
+      - gfortran
+      - pcre-dev
+      # libopenblas-dev liblapack-dev not sure if these are needed, builds fine without.  From https://github.com/kubeflow/katib/blob/b9dc63efb55bec19bdd80ca6f2930a65795f7c41/cmd/earlystopping/medianstop/v1beta1/Dockerfile#L10
+
+data:
+  - name: go-builds
+    items:
+      cert-generator: cert-generator/v1beta1
+      katib-controller: katib-controller/v1beta1
+      db-manager: db-manager/v1beta1
+      metricscollector: metricscollector/v1beta1/file-metricscollector
+      suggestion-goptuna: suggestion/goptuna/v1beta1
+
+  - name: python-builds
+    items:
+      earlystopping: earlystopping/medianstop/v1beta1
+      metricscollector: metricscollector/v1beta1/tfevent-metricscollector
+      suggestion-hyperband: suggestion/hyperband/v1beta1
+      suggestion-hyperopt: suggestion/hyperopt/v1beta1
+      suggestion-nas-darts: suggestion/nas/darts/v1beta1
+      suggestion-nas-enas: suggestion/nas/enas/v1beta1
+      suggestion-optuna-enas: suggestion/optuna/v1beta1
+      suggestion-pbt-enas: suggestion/pbt/v1beta1
+      suggestion-skopt-enas: suggestion/skopt/v1beta1
+
+pipeline:
+  - uses: git-checkout
+    with:
+      repository: https://github.com/kubeflow/katib
+      expected-commit: 05ab6f012ab6723609700a47a6cfa1845080e9b6
+      tag: v${{package.version}}
+
+subpackages:
+  - range: go-builds
+    name: "katib-${{range.key}}"
+    description: Kubeflow Katib ${{range.key}}
+    pipeline:
+      - uses: go/build
+        with:
+          packages: "./cmd/${{range.value}}"
+          output: katib-${{range.key}}
+          ldflags: -X main.version=${{package.version}}
+          subpackage: true
+
+  - range: python-builds
+    name: katib-${{range.key}}
+    description: Kubeflow Katib ${{range.key}}
+    pipeline:
+      - runs: |
+          cd ./cmd/${{range.value}}
+          pip install --prefer-binary -r requirements.txt --prefix=/usr --root=${{targets.subpkgdir}}
+
+          install -Dm755 ./main.py ${{targets.subpkgdir}}/usr/bin/
```

Comment:
So one thing about this is that we won't be able to install `katib-a` and `katib-b` in the same image, since they'll conflict on which gets to provide `/usr/bin/main.py`. Should we name them uniquely to avoid this?

Reply:
yeah good point!

Reply:
looking into this a bit more - I wonder where we should put these python builds, should this even go into `/usr/bin`. upstream they use `/opt` but for wolfi we avoid using this folder. 

additionally the auto diff comment failed as the message body failed to POST to the github comment.  A manual tree for one of the python related packages shows a huge number of files https://gist.github.com/rawlingsj/0102e404248ba1381153900997b2c192

so I think there may be something else wrong with these python based subpackages.

After looking at the upstream image

```
docker run -ti --entrypoint bash kubeflowkatib/tfevent-metrics-collector:latest
```
I can see the core python related files are written to `/usr/local/bin/`

the main.py is indeed namespaced in a folder 
```
/opt/katib/cmd/metricscollector/v1beta1/tfevent-metricscollector# ls
Dockerfile  Dockerfile.ppc64le	main.py  requirements.txt
```
in fact `/opt/katib/` seems to contain the whole katib git repo.